Change assert --> CUDA_ASSERT_KERNEL to avoid hip undefined __assert_fail

xw285cornell opened the pull request:
Summary:
Change assert --> CUDA_ASSERT_KERNEL to avoid hip undefined __assert_fail()

Otherwise crash trace:

```
caffe2/caffe2/operators/hip/top_k_radix_selection_hip.cuh:409:7: error:  '__assert_fail':  no overloaded function has restriction specifiers that are compatible with the ambient context 'gatherTopK'
      assert(writeIndex < outputSliceSize);
      ^
glibc/include/assert.h:88:6: note: expanded from macro 'assert'
   : __assert_fail (#expr, __FILE__, __LINE__, __ASSERT_FUNCTION))
     ^
```

Differential Revision: D13042820

bddppq commented on caffe2/operators/top_k_radix_selection.cuh:
```diff
@@ -1,6 +1,7 @@
 #ifndef CAFFE2_OPERATORS_TOP_K_RADIX_SELECTION_H_
 #define CAFFE2_OPERATORS_TOP_K_RADIX_SELECTION_H_
 
+#include "caffe2/core/context_gpu.h"
```
better to just include common_gpu.h

bddppq commented:
@pytorchbot retest this please

xw285cornell committed:
Change assert --> CUDA_ASSERT_KERNEL to avoid hip undefined __assert_fail (#13902)

Summary:
Change assert --> CUDA_ASSERT_KERNEL to avoid hip undefined __assert_fail()

Otherwise crash trace:

```
caffe2/caffe2/operators/hip/top_k_radix_selection_hip.cuh:409:7: error:  '__assert_fail':  no overloaded function has restriction specifiers that are compatible with the ambient context 'gatherTopK'
      assert(writeIndex < outputSliceSize);
      ^
glibc/include/assert.h:88:6: note: expanded from macro 'assert'
   : __assert_fail (#expr, __FILE__, __LINE__, __ASSERT_FUNCTION))
     ^
```
Pull Request resolved: https://github.com/pytorch/pytorch/pull/13902

Reviewed By: bddppq

Differential Revision: D13042820

Pulled By: bddppq

fbshipit-source-id: 53780e306719631247bc10384ef7ef0be631fdab
